src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -186,6 +181,16 @@ private static void EmitRegexMethod(IndentedTextWriter writer, RegexMethod rm, s
             // Main implementation methods
             writer.WriteLine($"            protected override void InitTrackCount() => base.runtrackcount = {rm.Code.TrackCount};");
             writer.WriteLine();
+#if DEBUG
+            writer.WriteLine("            // Node tree:");
```

Comment:
unrelated, this file has over 400 instances of `writer.WriteLine(...)`. could we have a method name say `wl` (or `writeLine` if must). either by making `writer` a field, or if not, then by having a nested method `wl(string s) => writer.WriteLine(s);`

seems like that would make it easier to read the emitted code.

Reply:
I'm not personally excited at that idea; you end up with lots of w(...) and wl(...) that are hard to see amongst everything else going on. There are also lots of places where writer is passed around,  e.g. to EmitBlock, and it'd look strange to me to use writer in one place but not in another. It's one of the things I actually find really annoying about the existing RegexCompiler code... it has all these shortcut methods that I need to learn and know what they mean rather than just calling the relevant methods on the ILGenerator instance. 

That's just me,  though.  I don't feel super strongly about it,  but I'm also not going to change it in this PR.

Reply:
Fair enough. If and when I ever hack on this file, I'll see whether it's an improvment.